base/MAPL_SwathGridFactory.F90

```diff
@@ -0,0 +1,1259 @@
+
+   
+   subroutine generate_newnxy(this,unusable,rc)
+      use MAPL_BaseMod, only: MAPL_DecomposeDim
+      class (SwathGridFactory), target, intent(inout) :: this
+      class (KeywordEnforcer), optional, intent(in) :: unusable
+      integer, optional, intent(out) :: rc
+      integer :: n
+
+      _UNUSED_DUMMY(unusable)
+
+      n = this%im_world/this%nx
+      if (n < 2) then
+         this%nx = generate_new_decomp(this%im_world,this%nx)
+         deallocate(this%ims)
+         allocate(this%ims(0:this%nx-1))
+         call MAPL_DecomposeDim(this%im_world, this%ims, this%nx)
+      end if
+      n = this%jm_world/this%ny
+      if (n < 2) then
+         this%ny = generate_new_decomp(this%jm_world,this%ny)
+         deallocate(this%jms)
+         allocate(this%jms(0:this%ny-1))
+         call MAPL_DecomposeDim(this%jm_world, this%jms, this%ny)
+      end if
+
+      _RETURN(_SUCCESS)
+
+   end subroutine generate_newnxy
+
+   function generate_new_decomp(im,nd) result(n)
+      integer, intent(in) :: im, nd
+      integer :: n
+      logical :: canNotDecomp
+
+      canNotDecomp = .true.
+      n = nd
+      do while(canNotDecomp)
+         if ( (im/n) < 2) then
+            n = n/2
+         else
+            canNotDecomp = .false.
+         end if
+      enddo
+   end function generate_new_decomp
+
+   subroutine init_halo(this, unusable, rc)
+      class (SwathGridFactory), target, intent(inout) :: this
+      class (KeywordEnforcer), optional, intent(in) :: unusable
+      integer, optional, intent(out) :: rc
+      STOP 'Stop: subroutine init_halo is not needed for SwathGridFactory'
+   end subroutine init_halo
+
+   subroutine halo(this, array, unusable, halo_width, rc)
+      use MAPL_CommsMod
+      class (SwathGridFactory), intent(inout) :: this
+      real(kind=REAL32), intent(inout) :: array(:,:)
+      class (KeywordEnforcer), optional, intent(in) :: unusable
+      integer, optional, intent(in) :: halo_width
+      integer, optional, intent(out) :: rc
+      STOP 'Stop: subroutine halo is not needed for SwathGridFactory'
+   end subroutine halo
+
+
+
+   subroutine append_metadata(this, metadata)
+      use MAPL_Constants
+      class (SwathGridFactory), intent(inout) :: this
+      type (FileMetadata), intent(inout) :: metadata
+
+      type (Variable) :: v
+      real(kind=REAL64), allocatable :: temp_coords(:)
+
+      ! Horizontal grid dimensions
+      call metadata%add_dimension('lon', this%im_world)
+      call metadata%add_dimension('lat', this%jm_world)
+
+      ! Coordinate variables
+      v = Variable(type=PFIO_REAL64, dimensions='lon,lat')
+      call v%add_attribute('long_name', 'longitude')
+      call v%add_attribute('units', 'degrees_east')
+      call metadata%add_variable('lons', v)
+
+      v = Variable(type=PFIO_REAL64, dimensions='lon,lat')
+      call v%add_attribute('long_name', 'latitude')
+      call v%add_attribute('units', 'degrees_north')
+      call metadata%add_variable('lats', v)
```

Comment:
I'm fairly certain that this is not the right metadata for swath.  But we can fix it later.  It should have along and across dimensions.  Not "lon" and "lat" dimensions.

Reply:
Right, we havenot yet introduced the along and across path directions, instead we are using lat/lon in a regular format.